src/history.cpp

```diff
@@ -1778,6 +1800,25 @@ void history_sanity_check() {
     // No sanity checking implemented yet...
 }
 
+wcstring history_session_id() {
+    wcstring result;
+
+    const env_var_t session_id = env_get_string(L"FISH_HISTFILE");
+    if (session_id.missing() || session_id.compare(L"default") == 0) {
+        result = L"fish";
+    } else if (session_id.empty()) {
+        result = L"";
+    } else if (valid_var_name(session_id)) {
+        result = session_id;
+    } else {
+        // TODO: the session id is invalid. Print an error.
+		debug(1, "TODO ERROR: Wrong characters in the session ID: '%ls'. Falling back to `fish`.", session_id.c_str());
```

Comment:
Also, I would restructure this for the common case which is to use the default session ID. Start with `wcstring result = L"fish";` and only set it to something else if the var has been set and its value is valid.